packages/safe-ds-vscode/src/extension/pythonServer.ts

```diff
@@ -297,17 +309,21 @@ export const executePipeline = async function (services: SafeDsServices, pipelin
     let codeMap: ProgramCodeMap = {};
     for (const generatedDocument of generatedDocuments) {
         const fsPath = URI.parse(generatedDocument.uri).fsPath;
-        lastGeneratedSource.set(fsPath, generatedDocument.getText());
-        if (fsPath.endsWith('.map')) {
-            // exclude sourcemaps
-            continue;
-        }
+        const workspaceRelativeFilePath = path.relative(workspaceRoot, path.dirname(fsPath));
         const sdsFileName = path.basename(fsPath);
         const sdsNoExtFilename =
             path.extname(sdsFileName).length > 0
                 ? sdsFileName.substring(0, sdsFileName.length - path.extname(sdsFileName).length)
                 : sdsFileName;
-        const workspaceRelativeFilePath = path.relative(workspaceRoot, path.dirname(fsPath));
+
+        lastGeneratedSource.set(
+            path.join(workspaceRelativeFilePath, sdsFileName).replaceAll('\\', '/'),
+            generatedDocument.getText(),
+        );
+        if (fsPath.endsWith('.map')) {
+            // exclude sourcemaps from sending to runner
+            continue;
+        }
```

Comment:
Is it possible to move this guard condition right behind the declaration of `fsPath` (i.e. line 312)? If its possible, move it please. Otherwise, add a comment why it must stay here.

Reply:
If it was moved, the sourcemaps would not be stored in the pipeline context anymore. Errors would no longer be mapped. 
I'll add a comment